esmvalcore/_recipe.py

```diff
@@ -348,6 +348,8 @@ def _get_default_settings(variable, config_user, derive=False):
 
     # Configure saving cubes to file
     settings['save'] = {'compress': config_user['compress_netcdf']}
+    if variable['short_name'] != variable['original_short_name']:
+        settings['save']['alias'] = variable['short_name']
```

Comment:
Wouldn't it be better to change the name when loading the cube instead of when saving?

Reply:
That would imply modifying the check to allow alternative names. It would also be confusing as we will use the modified short_name to report all the error

Reply:
Thinking about it some more, is it really necessary to change the variable name from the original?

Reply:
Only if we want the diagnostic to get all with the same `short_name`, and I think we do

Reply:
Why?

Reply:
That would be the short name provided by the yaml interface file, but I guess you're right and it somehow makes sense to keep this in line with the name used inside the file. We do not do this for any of the other attributes provided by the interface though.